src/app/static/src/app/components/plots/barchart/Barchart.vue

```diff
@@ -1,42 +1,46 @@
 
 <template>
     <div>
-        <div class="row">
+        <div v-if="initialised" class="row">
             <div class="col-md-3">
                 <div id="indicator-fg" class="form-group">
                     <label class="font-weight-bold">Indicator</label>
                     <treeselect :multiple=false
                                 :clearable="false"
                                 :options="indicators"
-                                v-model=indicatorId
+                                :value=selections.indicatorId
+                                @input="changeIndicatorId"
                                 :normalizer="normalizeIndicators"></treeselect>
                 </div>
                 <div id="x-axis-fg" class="form-group">
                     <label class="font-weight-bold">X Axis</label>
                     <treeselect :multiple=false
                                 :clearable="false"
                                 :options="filtersAsOptions"
-                                v-model=xAxisId></treeselect>
+                                :value=selections.xAxisId
+                                @input="changeXAxisId"></treeselect>
                 </div>
                 <div id="disagg-fg" class="form-group">
                     <label class="font-weight-bold">Disaggregate by</label>
                     <treeselect :multiple=false
                                 :clearable="false"
                                 :options="filtersAsOptions"
-                                v-model=disaggregateById></treeselect>
+                                :value=selections.disaggregateById
```

Comment:
What do you think about sticking with `v-models` for these controls but just using computed props with getters/setters? 

E.g.
```
            disaggregateById: {
                get() {
                    return this.selections.disaggregateById
                },
                set(value: string) {
                    this.$emit("change-selections", {disaggregateById: value});
                }
            }
```

The `created` handler would become a bit simpler e.g.

```
if (!this.indicatorId) {
    this.indicatorId = this.indicators[0].indicator;
}

if (!this.xAxisId) {
    this.xAxisId = this.filters[0].id;
}
...etc.
```

Reply:
OK, sounds good - the only thing with that is getting a flurry of emitted changes on initialise - but that will largely go away once we're getting defaults from the API.